documentation/binding-data/tutorial-flow-data-provider.asciidoc

```diff
@@ -1,50 +1,109 @@
+You can use basic components such as HTML elements to display such lists,
+or you can use components especially designed for the purpose, such as
+`Grid`, `ComboBox`, and `ListBox`.
 
-A listing component is a component that:
+[source, java]
+----
+// Create a listing component for a bean type
+Grid grid = new Grid(Person.class);
 
-* Displays one or several properties from a list of items, 
-* Allows the user to inspect the data and mark items as selected, and 
-* Optionally, allows the user to edit items directly in the component.
+// Sets items using vararg beans
+grid.setItems(
+        new Person("George Washington", 1732),
+        new Person("John Adams", 1735),
+        new Person("Thomas Jefferson", 1743),
+        new Person("James Madison", 1751)
+);
+----
 
-There are a number of officially-supported listing components, such as `ListBox` and `ComboBox`. Each component has its own API to configure exactly how the data is represented and manipulated. 
+All listing components in Vaadin have a number of `setItems()` methods to define the items to display.
+Items can be basic objects, such as strings or numbers, or POJOs, such as DTOs or JPA entities, from your domain model.
+The easiest way is to provide a `java.util.List` of objects to be shown in such a component.
 
-All list components have a `setItems` method to define which items display. They also have the `DataProvider` interface for more fine-grained control of the displayed data.
+If the number of items is large and reserves a lot of memory, `Grid` and `ComboBox` allow you to do a lazy data binding using callbacks to fetch only the required set of items from the back-end.
 
-== Changing How Items Display
+== Configuring How Items Are Displayed
 
-By default, most components use the `toString()` method to display items. If this is not suitable, you can change the behavior by configuring the component. Components are configured with one or more callbacks that define how to display the items.
+Component-specific APIs allow adjusting how items are displayed.
+Many listing components use the `toString()` method to display items by default.
+If this is not suitable, you can change the behavior by configuring the component.
+Listing components have one or more callbacks that define how to display the items.
 
-*Example*: A `ComboBox` component that lists status items and uses the `Status.getLabel()` method to represent each status. There is also a `Grid` with two columns, Name and Year of birth.
+For example, consider the `ComboBox` component that lists status items.
+You can configure it to use `Status.getLabel()` method to get a label for each status item.
 
 [source, java]
 ----
 ComboBox<Status> comboBox = new ComboBox<>();
 comboBox.setItemLabelGenerator(Status::getLabel);
+----
+
+In a `Grid`, you can use `addColumn()` to define the columns and configure the getter that returns the content for the column.
```

Comment:
The brackets are emphasise this is a method, so I think here it's fine. Don't want to use `Grid::` here, because of `Grid` word repetitions may waste the sentence.

Reply:
The word `method` should be used to point that this is a method.
This is a documentation and it should be absolutely clear without extra symbols which probably does something.
Even though the word `method`  adds one more word it's better to do it because this is not a code but just a text.

Reply:
Up to you.